### PR TITLE
Add tools_update_repo for Debian systems

### PR DESCRIPTION
**Pull request by srbarrios**
## What does this PR change?

We noticed that we don't inject anywhere the tools_update_repo for Debian systems.

### CODE REVIEW

File: salt/repos/default.sls [hunk 1]
```diff
@@ -727,6 +727,13 @@ wait_until_apt_lock_file_unlock:
       - interval: 5
       - until: True
 
+{% endif %} {# grains['os_family'] == 'Debian' #}
```
Comment (nodeg) [inline salt/repos/default.sls:730]: Doesn't this comment break the Jinja templating?

Comment (nodeg) [reply salt/repos/default.sls:730]: Ok, according to https://jinja.palletsprojects.com/en/3.0.x/templates/#comments

```jinja
{# ... #} 
```
> for [Comments](https://jinja.palletsprojects.com/en/3.0.x/templates/#comments) not included in the template output

is fine.

File: salt/repos/default.sls [hunk 2]
```diff
@@ -792,7 +792,71 @@ tools_update_repo_raised_priority:
...
+
+{% if grains['os'] == 'Debian' %}
+
+{% set release = grains.get('osrelease', None) %}
+
+tools_update_repo:
+  pkgrepo.managed:
+    - humanname: tools_update_repo
+    - file: /etc/apt/sources.list.d/tools_update_repo.list
+    # We only have one shared Client Tools repository, so we are using 4.3 for 4.2
```
Comment (nodeg) [inline salt/repos/default.sls:805]: This comment needs adjustment.

Comment (nodeg) [reply salt/repos/default.sls:805]: Furthermore we have no entry for 5.0?

File: salt/repos/default.sls (hunk 1 above)
Comment (srbarrios) [reply salt/repos/default.sls:730]: Yes, if you notice, we have it for every of these sections in the rest of family names, it helps us to navigate through this sls.

File: salt/repos/default.sls (hunk 2 above)
Comment (srbarrios) [reply salt/repos/default.sls]: Fixed, being generic, to don't need to adjust it next time.
For now, I think we only have Head right?

**Review by nodeg — APPROVED** [review]
Overall this LGTM. There is only the change, Maxime wanted to propose. But with this PR merged before, we can merge his PR as is.

**Review by vandabarata — APPROVED** [review]

**Review by Bischoff — APPROVED** [review]

**Review by NamelessOne91 — APPROVED** [review]

**Review by maximenoel8 — APPROVED** [review]

**Review by nodeg — APPROVED** [review]

**Pull request merged**